src/sdk/namespace_fs.js

```diff
@@ -333,7 +333,11 @@ class NamespaceFS {
             const file_path = this._get_file_path(params);
             const stat = await fs.promises.stat(file_path);
             console.log(file_path, stat);
-            return this._get_object_info(params.bucket, params.key, stat);
+            if (stat.isDirectory()) {
+                const err = new Error('NoSuchKey');
+                err.code = 'ENOENT';
+                throw this._translate_object_error_codes(err);
+            }
```

Comment:
@eshelmarc you need to add back this line:
```
return this._get_object_info(params.bucket, params.key, stat);
```

Reply:
basically you need to push your new if block before the existing return get object info line, but keep it after your new block.